### ISSUE TITLE
UnicodeEncodeError

### ISSUE THREAD

**Issue by foocp** [haccer/tweep#24]
I always get this error:
`UnicodeEncodeError: 'ascii' codec can't encode character '\xfc' in position 76: ordinal not in range(128)
`

I'm on the latest Raspbian Stretch 
> Linux raspberrypi 4.9.59-v7+ #1047 SMP Sun Oct 29 12:19:23 GMT 2017 armv7l

using
> Python 3.5.3 (default, Jan 19 2017, 14:11:04) 
> [GCC 6.3.0 20170124] on linux

locale is:
> locale: Cannot set LC_CTYPE to default locale: No such file or directory
> locale: Cannot set LC_ALL to default locale: No such file or directory
> LANG=en_GB.UTF-8
> LANGUAGE=
> LC_CTYPE=UTF-8
> LC_NUMERIC="en_GB.UTF-8"
> LC_TIME="en_GB.UTF-8"
> LC_COLLATE="en_GB.UTF-8"
> LC_MONETARY="en_GB.UTF-8"
> LC_MESSAGES="en_GB.UTF-8"
> LC_PAPER="en_GB.UTF-8"
> LC_NAME="en_GB.UTF-8"
> LC_ADDRESS="en_GB.UTF-8"
> LC_TELEPHONE="en_GB.UTF-8"
> LC_MEASUREMENT="en_GB.UTF-8"
> LC_IDENTIFICATION="en_GB.UTF-8"
> LC_ALL=

**Comment by haccer** [comment]
Ok, I’m going to try reproducing this.

**Comment by pachonk** [comment]
This is a fairly common python issue, first piece on the Unicode docs here
I believe https://docs.python.org/2.7/howto/unicode.html#the-unicode-type

Essentially instead of calling str, you have to call encode instead since
str assumes ascii and it's really utf-8

On mobile and replying in email, I forgot if this was my 2 or 3. Docs
should be pretty much the same for either on Unicode I thought though.

**Comment by haccer** [comment]
Yea it appears when I do a search of ü which is \xfc, tweep won't scrape tweets. I tested this with some other unique unicode characters and got the same result. Going to try and figure out a fix next week.

**Comment by tredmill** [comment]
Thank you for the script Haccer. We changed the script so that it does not stop, but it is by no means an elegant solution. This might work for those in a hurry until you find a more sustainable alternative. 

`async def getTweets(init):
    tweets, init = await getFeed(init)
    count = 0
    for tweet in tweets:
        try:
            tweetid = tweet["data-item-id"]
            datestamp="01 jan 1970"
            try:
                datestamp = tweet.find("a", "tweet-timestamp")["title"].rpartition(" - ")[-1]
            except TypeError:
                pass
            d = datetime.datetime.strptime(datestamp, "%d %b %Y")
            date = d.strftime("%Y-%m-%d")

            timestamp="00:00:00"
            try:
                timestamp = str(datetime.timedelta(seconds=int(tweet.find("span", "_timestamp")["data-time"]))).rpartition(", ")[-1]
            except TypeError:
                pass        
            t = datetime.datetime.strptime(timestamp, "%H:%M:%S")
            time = t.strftime("%H:%M:%S")
            username = tweet.find("span", "username").text.replace("@", "")
            timezone = strftime("%Z", gmtime())
            text = tweet.find("p", "tweet-text").text.replace("\n", " ").replace("http"," http").replace("pic.twitter"," pic.twitter")
            hashtags = ",".join(re.findall(r'(?i)\#\w+', text, flags=re.UNICODE))
            replies = tweet.find("span", "ProfileTweet-action--reply u-hiddenVisually").find("span")["data-tweet-stat-count"]
            retweets = tweet.find("span", "ProfileTweet-action--retweet u-hiddenVisually").find("span")["data-tweet-stat-count"]
            likes = tweet.find("span", "ProfileTweet-action--favorite u-hiddenVisually").find("span")["data-tweet-stat-count"]
            try:
                mentions = tweet.find("div", "js-original-tweet")["data-mentions"].split(" ")
                for i in range(len(mentions)):
                    mention = "@{}".format(mentions[i])
                    if mention not in text:
                        text = "{} {}".format(mention, text)
            except:
                pass
            if arg.users:
                output = username
            elif arg.tweets:
                output = tweets
            else:
                output = "{} {} {} {} <{}> {}".format(tweetid, date, time, timezone, username, text)
                if arg.hashtags:
                    output+= " {}".format(hashtags)
                if arg.stats:
                    output+= " | {} replies {} retweets {} likes".format(replies, retweets, likes)

            if arg.o != None:
                if arg.csv:
                    dat = [tweetid, date, time, timezone, username, text, hashtags, replies, retweets, likes]
                    with open(arg.o, "a", newline='') as csv_file:
                        writer = csv.writer(csv_file, delimiter="|")
                        writer.writerow(dat.encode('utf-8'))
                else:
                    print(output.encode('utf-8'), file=open(arg.o, "a"))

            count += 1
            print(output)
        except:
            pass`

**Comment by haccer** [comment]
Sorry it took until today for me to replicate your environment.

**tl;dr**
The problem was an issue with your locale. To fix, simply run:
```
export LC_CTYPE=en_GB.UTF-8
```

--
I tested w/ the default locale against Twitter's that contained special characters and had no issues, but after changing to your specific locale settings I got that error. This kinda threw me off a little because, Python3 is supposed to be unicode by default.

After exporting that variable, I confirmed it worked and was not an issue with Tweep itself or connected to the known-issue of searching special characters with Tweep.

**Comment by pachonk** [comment]
Had to be British not us?

**Comment by haccer** [comment]
@pachonk nah, his locale had: `LC_CTYPE=UTF-8`

I guess language has to be specified, so for USA it'd be `en_US.UTF-8`

**Comment by pachonk** [comment]
Okay, I thought so but I was curious if only the local for GB worked.
Thanks!

**Comment by pachonk** [comment]
Okay, I thought so but I was curious if only the local for GB worked.
Thanks!
